packages/rs-dpp/src/identity/identity_public_key/in_creation.rs

```diff
@@ -0,0 +1,156 @@
+use crate::identity::{IdentityPublicKey, KeyID, KeyType, Purpose, SecurityLevel};
+use ciborium::value::Value as CborValue;
+use std::convert::TryInto;
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde_json::Value as JsonValue;
+
+use crate::errors::ProtocolError;
+use crate::util::cbor_value::{CborCanonicalMap, CborMapExtension};
+use crate::util::json_value::{JsonValueExt, ReplaceWith};
+use crate::SerdeParsingError;
+
+pub const BINARY_DATA_FIELDS: [&str; 2] = ["data", "signature"];
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct IdentityPublicKeyInCreation {
+    pub id: KeyID,
+    pub purpose: Purpose,
+    pub security_level: SecurityLevel,
+    #[serde(rename = "type")]
+    pub key_type: KeyType,
+    pub data: Vec<u8>,
+    pub read_only: bool,
+    /// The signature is needed for ECDSA_SECP256K1 Key type and BLS12_381 Key type
+    pub signature: Vec<u8>,
+}
+
+impl IdentityPublicKeyInCreation {
```

Comment:
This structure belongs to State Transition so we should move it `state_transition` module like we have `DocumentCreateTransition`, `DocumentDeleteTransition` sub-structures. Also, it could be called similarly like `IdentityPublicKeyAddTransition`

Reply:
Let's do this later.

Reply:
If we don't do it now we never do it, as usual.